server/build.go

```diff
@@ -299,10 +299,10 @@ func (task *BuildTask) build(tracing *stringSet) (esm *ESM, err error) {
 					}
 
 					// bundles all dependencies in `bundle` mode, apart from peer dependencies
-					if task.BundleMode && !extraExternal.Has(specifier) {
+					if task.BundleMode && !extraExternal.Has(specifier) && !task.external.Has(specifier) {
```

Comment:
possibly, better to use pkgNameInfo.fullname

Reply:
i guss use `spcifier` is ok